walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,55 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.stream.*;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    //Approach: start form start of day, and add available timeranges as we traverse through the events' timeranges
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        Collection<TimeRange> validTimeRanges = new ArrayList();
+        long duration = request.getDuration();
+        int start = 0;
+        int end = 0;
+        int lastEnd = 0;
+
+        // Get all the event time ranges and sort them by the start time. 
+        // Only add to the list if at least one attendee from the event is 
+        //a required attendee in the meeting request. (Otherwise, the event is irrelevant.)
+        List<TimeRange> eventTimeRanges = events.stream().filter(
+         event -> event.getAttendees().stream().anyMatch(
+             attendee -> request.getAttendees().contains(attendee)))
+         .map(e -> e.getWhen())
+         .collect(Collectors.toList());
```

Comment:
nit: continued lines should indent by 4: https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent

Reply:
Done!